nomenclature/processor/region.py

```diff
@@ -390,18 +392,17 @@ def validate_with_definition(cls, v, values):
         v.validate_regions(values["region_codelist"])
         return v
 
-    def apply(self, df: IamDataFrame) -> IamDataFrame:
+    def apply(self, df: IamDataFrame) -> Tuple[IamDataFrame, pd.DataFrame]:
         """Apply region processing
 
         Parameters
         ----------
         df : IamDataFrame
             Input data that the region processing is applied to
-
         Returns
         -------
-        IamDataFrame
-            Processed data
+        Tuple[IamDataFrame, pd.DataFrame]:
```

Comment:
If we stick with this approach, please extend the docstring to show that the return-type depends on the configuration.